server/notification-providers/whapi.js

```diff
@@ -0,0 +1,46 @@
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+
+class Whapi extends NotificationProvider {
+
+    name = "whapi";
+
+    /**
+     * @inheritdoc
+     */
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+
+        let okMsg = "Sent Successfully.";
+
+        let apiUrl = notification.whapiApiUrl;
+        let apiToken = notification.whapiAuthToken;
+        let toNumber = notification.whapiToNumber;
+
+        try {
+
+            let config = {
+                headers: {
+                    "Accept": "application/json",
+                    "Content-Type": "application/json",
+                    "Authorization": "Bearer " + apiToken,
+                }
+            };
+
+            let data = {
+                "to": toNumber + "@s.whatsapp.net",
```

Comment:
> pattern: `^[\d-]{10,31}(@[\w\.]{1,})?$`
example: `1234567891@s.whatsapp.net`
Use the phone number or Chat ID of the contact/group to which you want to send the message.
Use [Get groups](https://whapi.readme.io/reference/getgroups) to get the group ID.

Looking at the api docs, it might be interesting to not append the `@s.whatsapp.net` here as the pattern does not require it.
I don't know enough about whatsapp to know which format a group ID is.

Reply:
I removed that `@s.whatsapp.net`, now it works in every case:
1. Number only (only digit, with international prefix)
2. Contact ID (number+`@s.whatsapp.net`)
3. Group ID (`123456789012345678@g.us`)